### PR TITLE
sile: 0.14.17 -> 0.15.5

### PR DESCRIPTION
**Pull request by doronbehar**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/si/sile/package.nix
Comment (doronbehar) [inline pkgs/by-name/si/sile/package.nix:44]: It is natural to put these close to each other, as the hashes will be updated together and hence the version bump diffs will be easier to review.

File: pkgs/by-name/si/sile/package.nix
Comment (doronbehar) [inline pkgs/by-name/si/sile/package.nix:4]: Almost for any package these come first.

File: pkgs/by-name/si/sile/package.nix
Comment (doronbehar) [inline pkgs/by-name/si/sile/package.nix:56]: Corresponding to the `lua` argument right after the `# buildInputs` comment at the top.

File: pkgs/by-name/si/sile/package.nix
Comment (doronbehar) [inline pkgs/by-name/si/sile/package.nix:65]: A "buildInput".

File: pkgs/by-name/si/sile/package.nix
Comment (doronbehar) [inline pkgs/by-name/si/sile/package.nix:94]: In many packages, `configureFlags` do something related to the `outputs`, so it is natural to put these right after `configureFlags`.

**Review by doronbehar — COMMENTED** [review]
Here are some explanations for why lists are ordered the way they are. I'd recommend to use the same orders in upstream's `pkg.nix` @alerque.

**Comment by doronbehar** [comment]
Waiting @alerque for your response on where to put the manual. I suggest copying it manually to `$out/share/doc/sile/manual.pdf`, but perhaps you'd like to put more files there from the `documentation` directory..

**Comment by alerque** [comment]
This arrangement makes sense to me and I'm happy to mirror it in the upstream Flake. I was just unhappy with the random collection of lists that were appended as-they-happened instead of in *some* logical order.

> Waiting @alerque for your response on where to put the manual. I suggest copying it manually to `$out/share/doc/sile/manual.pdf`, but perhaps you'd like to put more files there from the `documentation` directory..

It looks like running `install documentation/sile.pdf ...` manually is the only realistic way right now. It's basically the only documentation artifact I'd include for now. The `make install-pdf` target used to work but seems to have broken when I reorganized docs related targets for generated Rust API docs, Lua API docs, etc. I can fix that upstream, and also someday it might make sense to include those in a special derivation output and wrap `make install-html` or similar, but for now we just are not installing any documentation automatically. And I think that's fine for most distro packaging scenarios.

File: pkgs/by-name/si/sile/package.nix
Comment (alerque) [reply pkgs/by-name/si/sile/package.nix:4]: I'm a bit puzzled what the issue with ZST is here and why my version was tripping up. It seems to work fine [in the Libertinus font package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/libertinus/default.nix).

**Review by alerque — APPROVED** [review]

File: pkgs/by-name/si/sile/package.nix
Comment (doronbehar) [reply pkgs/by-name/si/sile/package.nix:4]: > I'm a bit puzzled what the issue with ZST is here and why my version was tripping up. It seems to work fine [in the Libertinus font package](https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/libertinus/default.nix?rgh-link-date=2024-11-04T11%3A17%3A44Z).

I know, unpacking zst tarballs is pretty much supported in Nixpkgs. The problem I experienced was with the `fetchCargoTarball` function that did not know how to handle those zst tarballs, giving out unclear errors..

**Comment by doronbehar** [comment]
> also someday it might make sense to include those in a special derivation output and wrap `make install-html` or similar

I agree, we can put those docs in a `"devdoc"` output when that is ready.

> It looks like running `install documentation/sile.pdf ...` manually is the only realistic way right now. It's basically the only documentation artifact I'd include for now.

Done.

**Comment by doronbehar** [comment]
Will merge when CI is green.

**Comment by doronbehar** [comment]
Feel free to ask me for a reviews regarding changes to Nix files upstream.

File: pkgs/by-name/si/sile/package.nix
Comment (alerque) [reply pkgs/by-name/si/sile/package.nix:4]: I'm glad I'm not the only one confused by that error! I wonder how hard that would be to fix (I'm sue the error handling is way above my pay grade, that's hard stuff, I'm just thinking of knowing how to uncompress a ztd file).

**Review by alerque — APPROVED** [review]

**Comment by doronbehar** [comment]
Just will note here before merging that I noticed a peculiar error on x86_64-darwin:

```
building '/nix/store/8w3w6mgw394sfwpkssg09bihcb19q7hx-sile-test.drv'...
SILE v0.15.5 (Lua 5.2) [Rust]
<STDIN> as xml
! Couldn't find face 'Gentium Plus'
 at     after STDIN: in <snippet>:
                [[<sile>foo</sile>␤]]
Error: runtime error: ...2s77c8blzmqv-sile-0.15.5/share/sile/shapers/harfbuzz.lua:95: 
stack traceback:
        [C]: in ?
        [C]: in function 'error'
        ...7c8blzmqv-sile-0.15.5/share/sile/core/utilities/init.lua:336: in function 'error'
        ...2s77c8blzmqv-sile-0.15.5/share/sile/shapers/harfbuzz.lua:95: in function 'callback'
        ...irh34x42s77c8blzmqv-sile-0.15.5/share/sile/core/font.lua:162: in function 'cache'
        ...2s77c8blzmqv-sile-0.15.5/share/sile/shapers/harfbuzz.lua:40: in function 'shapeToken'
        ...34x42s77c8blzmqv-sile-0.15.5/share/sile/shapers/base.lua:97: in function 'createNnodes'
        ...rh34x42s77c8blzmqv-sile-0.15.5/share/sile/types/node.lua:298: in function 'shape'
        ...2s77c8blzmqv-sile-0.15.5/share/sile/typesetters/base.lua:536: in function 'shapeAllNodes'
        ...77c8blzmqv-sile-0.15.5/share/sile/packages/bidi/init.lua:182: in function 'boxUpNodes'
        ...2s77c8blzmqv-sile-0.15.5/share/sile/typesetters/base.lua:947: in function 'leaveHmode'
        ...34x42s77c8blzmqv-sile-0.15.5/share/sile/classes/base.lua:716: in function 'endPar'
        ...2s77c8blzmqv-sile-0.15.5/share/sile/typesetters/base.lua:352: in function 'endline'
        ...34x42s77c8blzmqv-sile-0.15.5/share/sile/classes/base.lua:737: in function 'finish'
        ...irh34x42s77c8blzmqv-sile-0.15.5/share/sile/core/sile.lua:698: in function <...irh34x42s77c8blzmqv-sile-0.15.5/share/sile/core/sile.lua:697>
```

I know there is an issue with ofborg trying to build such `passthru.tests.test` derivations when it is not even supported on such platforms. On a real `aarch64-darwin` machine, I tried building `sile.passthru.tests.test` and it simply built nothing, so I tested that `<sile>foo</sile>` string on the resulting `sile` file, and it worked OK (besides the default font issue which does not worry me).

**Pull request merged**
